coded_tools/base_rag.py

```diff
@@ -13,24 +13,53 @@
 import re
 from abc import ABC
 from abc import abstractmethod
+from dataclasses import dataclass
 from typing import Any
 from typing import List
+from typing import Literal
 from typing import Optional
 import logging
 
+# pylint: disable=import-error
+from asyncpg import InvalidCatalogNameError
+from asyncpg import InvalidPasswordError
 from langchain_community.vectorstores import InMemoryVectorStore
 from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
 from langchain_core.vectorstores import VectorStore
 from langchain_core.vectorstores.base import VectorStoreRetriever
 from langchain_openai import OpenAIEmbeddings
+from langchain_postgres import PGEngine
+from langchain_postgres import PGVectorStore
 from langchain_text_splitters import RecursiveCharacterTextSplitter
+from sqlalchemy.exc import ProgrammingError
 
 # Invalid file path character pattern
 INVALID_PATH_PATTERN = r"[<>:\"|?*\x00-\x1F]"
+DEFAULT_TABLE_NAME = "vectorstore"
+EMBEDDINGS_MODEL = "text-embedding-3-small"
+VECTOR_SIZE = 1536
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class PostgresConfig:
+    """Configuration for PostgreSQL connection."""
```

Comment:
We try to keep one class per file that is named the same as the class itself so it can be found easily